internal/consensus/state.go

```diff
@@ -1360,6 +1360,17 @@ func (cs *State) proposalIsTimely() bool {
 	return cs.Proposal.IsTimely(cs.ProposalReceiveTime, sp)
 }
 
+func (cs *State) timelyProposalMargins() (time.Duration, time.Duration) {
```

Comment:
These margins are not adaptive according to the round, see [this pattern](https://github.com/cometbft/cometbft/pull/2431/files#diff-c27a58d5a8b3c15166169543b3cdb94da91bb60efa27609ef446719474af78c5R2696-R2702)

Reply:
Perhaps we don't export `SynchronyParams` as a type and instead only export the constructor function. Can follow up in a subsequent PR

Reply:
Yes, we need to adapt this once #2431 is merged.

Actually, let me make it the target branch.